sonic_y_cable/y_cable_base.py

```diff
@@ -0,0 +1,636 @@
+                 0bit for lane 0, 1bit for lane1 and so on.
+                 for example 3 -> 0b'0011 , means running on lane0 and lane1
+
+        Returns:
+            a boolean, True if the enable is successful
+                     , False if the enable failed
+        """
+
+        raise NotImplementedError
+
+    def disable_loopback_mode(self, target):
+        """
+        This API specifically disables the Loopback mode on the port user provides.
+        Target is an integer for selecting which end of the Y cable we want to run loopback on.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+            target:
+                One of the following predefined constants, the target on which to disable the loopback:
+                    EYE_PRBS_TARGET_LOCAL -> local side,
+                    EYE_PRBS_TARGET_TOR_A -> TOR A
+                    EYE_PRBS_TARGET_TOR_B -> TOR B
+                    EYE_PRBS_TARGET_NIC -> NIC
+
+        Returns:
+            a boolean, True if the disable is successful
+                     , False if the disable failed
+        """
+
+        raise NotImplementedError
+
+    def enable_debug_mode(self):
+        """
+        This API specifically configures and enables a debug mode that the port is now
+        going to be run on. This means that PRBS/Loopback etc. type diagnostic mode
+        is now going to be run on the port and hence normal traffic will be disabled
+        on it.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+
+        Returns:
+            a boolean, True if the enable is successful
+                     , False if the enable failed
+        """
+
+        raise NotImplementedError
+
+    def disable_debug_mode(self):
+        """
+        This API specifically disables a debug mode that the port was currently being run on.
+        This means that PRBS/Loopback etc. type diagnostic mode
+        are now going to be stopped the port and hence normal traffic will be enabled
+        on it.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+
+        Returns:
+            a boolean, True if the disable is successful
+                     , False if the disable failed
+        """
+
+        raise NotImplementedError
+
+    def is_debug_mode_enabled(self):
+        """
+        This API specifically checks if a debug mode is currently being run on the port
+        for which this API is called for.
+        This means that PRBS/Loopback etc. type diagnostic mode
+        if any are being run on the port this should return True else False
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+
+        Returns:
+            a boolean, True if debug mode enabled
+                     , False if debug mode not enabled
+        """
+
+        raise NotImplementedError
```

Comment:
Suggest condensing these into two functions:

`set_debug_mode(bool)`
`get_debug_mode()`

Also suggest moving them to the top of the debug functionality section.

Reply:
fixed